docs/getting-started-guides/coreos/azure/lib/azure_wrapper.js

```diff
@@ -107,6 +107,9 @@ var create_ssh_key = function (prefix) {
   };
   openssl.exec('req', opts, function (err, buffer) {
     if (err) console.log(clr.red(err));
+    openssl.exec('rsa', { in: opts.keyout, out: opts.keyout }, function (err, buffer) {
+       if (err) console.log(clr.red(err));
+    });
     fs.chmod(opts.keyout, '0600', function (err) {
```

Comment:
I'm not a js programmer, but wouldn't the chmod need to go inside the callback for the second exec to operate on the correct file?

Can you verify that the file permissions are correct on the generated key after this change?

Adding @errordeveloper